source/computer/commands.rst

```diff
@@ -63,54 +62,496 @@ Linux/macOS 下有成百上千个命令，每个命令都有众多选项。这
 
     -   其他
 
+        - `df`_
+        - `du`_
         - `ssh`_
         - `scp`_
         - `rsync`_
+        - `wget`_
+        - `top`_
 
 ----
 
 cat
 ---
 
+``cat`` 命令的命名来源于 con\ **cat**\ enate（即“拼接”）。该命令可以输出某个文件的
+内容，或将多个文件\ **拼接**\ 起来::
+
+    # 将一个文件的内容输出到终端
+    $ cat file
+    
+    # 输出文件内容并显示行号
+    $ cat -n file
+    
+    # 将文件 file1 和 file2 合并到一个文件 target_file 中
+    $ cat file1 file2 > target_file
+
+diff
+----
+
+``diff`` 命令的命名来自 **diff**\ erence。该命令可以用来逐行比较文件的异同。
+
+我们用以下两个示例文件展示用法::
+
+    $ cat file1 
+    seismo-learn
+    seismology
+    software
+    
+    $ cat file2
+    seismo-learn
+    seismology101
+    software
+
+比较 :file:`file1` 和 :file:`file2` 的异同，结果显示两个文件第二行不同，其他
+行是相同的::
+
+    $ diff file1 file2
+    2c2
+    < seismology
+    ---
+    > seismology101
+
+df
+--
+
+``df`` 命令的命名来自 **d**\ isk **f**\ ree（可使用的磁盘空间）。该命令可以获取硬盘被占用了多少空间，
+目前还剩下多少空间等信息。
+
+::
+
+    $ df -h
+    Filesystem                          Size  Used Avail Use% Mounted on
+    /dev/mapper/cl_ivantjuawinata-root  100G   86G   15G  86% /
+    devtmpfs                            7.8G     0  7.8G   0% /dev
+    tmpfs                               7.8G  114M  7.7G   2% /run
+    /dev/mapper/cl_ivantjuawinata-home  1.1T  904G  149G  86% /home
+    
+从输出中可以得知，`/home` 分区总磁盘空间为1.1 TB，已使用 904 GB，剩余 149 GB。
+
+du
+--
+
+``du`` 命令的名字来自 **d**\ isk **u**\ sage（磁盘使用情况）。该命令可以查看磁盘的使用空间。
+但与 `df`_ 命令不同的是，\ ``du`` 命令用于查看文件和目录磁盘使用的空间。
+
+查看当前目录所占磁盘空间（也可以在最后指定想查看的目录名）::
+
+    # 默认显示当前目录及其所有子目录所占磁盘空间
+    $ du -h
+
+    # 设置目录深度为 0，即只显示当前目录所占磁盘空间，不显示其任何子目录
+    $ du -h -d 0
+
+    # 设置目录深度为 1，即只显示当前目录及其一级子目录所占磁盘空间
+    $ du -h -d 1
+
+查看指定文件所占磁盘空间::
+
+    $ du -h file1.txt
+
 find
 ----
 
+``find`` 命令用来查找指定目录下的子目录和文件，并显示查找到的子目录和文件::
```

Comment:
find is more powerful than the commands shown here. Should we add a few more?

1. find files and delete them
2. find files and run a command on these files

Reply:
I am not sure if we should use pipe here.

Reply:
I guess we can. Please show some examples cause I never use this command.

Reply:
No. It's not about pipes.

> * find files and delete them

`find . -name ".pyc" -delete`

> * find files and run a command on these files

`find . -name ".c" -exec grep "pattern" {} +`

Reply:
cool. I never kown it.

Reply:
`find` is very powerful and we may need to give a link of a good Chinese tutorial, like the gawk and curl ones.

Reply:
done in 98e5172.